### PR TITLE
Rename Parser2 to Parser, and delete the original

### PR DESCRIPTION
**Pull request by jonmeow**

### CODE REVIEW

File: toolchain/parser/parse_tree.h [hunk 1]
```diff
@@ -166,7 +159,7 @@ class ParseTree {
                       int subtree_size_arg)
         : kind(k), token(t), subtree_size(subtree_size_arg) {}
 
-    // TODO: Parser2 only uses this construct. Can remove the other if we
+    // TODO: Parser only uses this construct. Can remove the other if we
```
Comment (chandlerc) [inline toolchain/parser/parse_tree.h]: Since this removes the other parser, remove the other here?

**Review by chandlerc — APPROVED** [review]
LGTM -- suggesting to GC one more bit of old parser.

File: toolchain/parser/parse_tree.h (hunk 1 above)
Comment (jonmeow) [reply toolchain/parser/parse_tree.h]: Thanks, removed

**Review by chandlerc — APPROVED** [review]
(still LGTM)

**Pull request merged**